sdk/appcenter-auth/src/main/java/com/microsoft/appcenter/auth/Auth.java

```diff
@@ -380,16 +380,22 @@ public void run() {
 
     @WorkerThread
     private synchronized void processDownloadedConfig(String payload, String eTag) {
+        boolean continueSignIn = isPendingSignInWaitingForConfiguration();
         mGetConfigCall = null;
         saveConfigFile(payload, eTag);
         AppCenterLog.info(LOG_TAG, "Configure auth from downloaded configuration.");
         initAuthenticationClient(payload);
+        if (continueSignIn) {
+            selectSignInTypeAndSignIn(mLastSignInFuture);
+        }
     }
 
-
     private synchronized void processDownloadNotModified() {
         mGetConfigCall = null;
         AppCenterLog.info(LOG_TAG, "Auth configuration didn't change.");
+        if (isPendingSignInWaitingForConfiguration()) {
+            mLastSignInFuture.complete(new SignInResult(null, new IllegalStateException("Cannot load auth configuration from the server.")));
```

Comment:
Should this be an `InternalServerErrorException` or similar?  Theoretically this is a problem with the CDN/server providing the config, so `IllegalStateException` doesn't seem quite right to me

Reply:
This is not an error case, and this cannot really happen to be delayed for that case.
If we ever get 304 that means we send an etag, meaning we had a config in the first place and thus we already allowed signIn.

I guess this is just a defensive check in case the impossible happens @jwallra.